x/protocolpool/keeper/keeper.go

```diff
@@ -397,19 +393,23 @@ func (k Keeper) calculateClaimableFunds(ctx context.Context, recipient sdk.AccAd
 	periodsPassed := int64(timeElapsed) / int64(*budget.Period)
 
 	// Calculate the amount to distribute for all passed periods
-	coinsToDistribute := math.NewInt(periodsPassed).Mul(budget.TotalBudget.Amount.QuoRaw(int64(budget.Tranches)))
+	coinsToDistribute := math.NewInt(periodsPassed).Mul(budget.BudgetPerTranche.Amount)
 	amount = sdk.NewCoin(budget.TotalBudget.Denom, coinsToDistribute)
 
 	// update the budget's remaining tranches
-	budget.TranchesLeft -= uint64(periodsPassed)
+	if budget.TranchesLeft > uint64(periodsPassed) {
+		budget.TranchesLeft -= uint64(periodsPassed)
+	} else {
+		budget.TranchesLeft = 0
+	}
 
 	// update the ClaimedAmount
 	claimedAmount := budget.ClaimedAmount.Add(amount)
 	budget.ClaimedAmount = &claimedAmount
 
 	// Update the last claim time for the budget
-	nextClaimFrom := budget.NextClaimFrom.Add(*budget.Period)
-	budget.NextClaimFrom = &nextClaimFrom
+	nextClaimFrom := budget.LastClaimedAt.Add(*budget.Period)
+	budget.LastClaimedAt = &nextClaimFrom
```

Comment:
Why is this not `budget.LastClaimedAt = currentTime`?

Reply:
we are considering the use case of fixed period between claims, i.e., if period is set to 3600 seconds, it should occur every hour. But if we set to currentTime, it may not reflect the intended periodic nature of claims.
If we want the next claim time to be a specific duration after the last claim time (i.e., having a fixed period between claims), then adding budget.Period to the last claimed time as done in the original logic makes sense.
We can set to currentTime but that wasn't our use case.
I can try to explain more if this doesn't make sense to you.

Reply:
Ah ok sounds good, in that case I believe we want to add `period * periodsPassed`, not just period?

Reply:
I think this is the only thing left for this to be merged

Reply:
I think you've addressed this 
ref: https://github.com/cosmos/cosmos-sdk/pull/20154/files#diff-dcf594a333e20d0c216d697c5b17b3a407ca56b2f8acd2f6f1e4a13904f5a513R419-R420

Reply:
lol you are right